python/lsst/ts/Dome/mock_controller.py

```diff
@@ -93,35 +95,30 @@ def __init__(
         await self.status()
         while True:
             print_ok = True
-            line = await reader.readline()
-            line = line.decode()
-            if not line:
-                # connection lost; close the writer and exit the loop
-                self.log.info("closing the writer")
-                writer.close()
-                self.log.info("stopping mock controller")
-                await self.stop()
-                return
-            line = line.strip()
-            self.log.info(f"read command: {line!r}")
+            line = await reader.readuntil("\r\n".encode())
+            line = line.decode().strip()
+            self.log.info(f"read command line: {line!r}")
             timeout = 20
             if line:
                 try:
-                    items = line.split(";")
-                    cmd = items[0]
+                    outputs = None
+                    items = yaml.safe_load(line)
+                    cmd = next(iter(items))
```

Comment:
I tried that and it doesn't work. A dictionary doesn't contain an iterator which is why I have to implicitly create one.

Reply:
Do you are assuming that the first item is a command? Now that dict is order-preserving I guess that works, but it is a rather surprising thing to do.

I don't know if it is worth trying to change, but I do think messages would be cleaner if the command and reply were available via fixed keys. e.g.:
* {"Command": <command>, ...parameters}
* {"Reply": "OK"/"Error"} vs {"OK": None} and {"Error: {"Code": 2}}

Reply:
Yes the first item is the command. Basically we went from "command;parameter=value" to "command:\n parameter: value\n" to be in line with yaml. Likewise, we went from "OK;Timeout=value" to "OK:\n Timeout: value" and from "ERROR;Code=2" to "ERROR:\n Code: 2\n". Yaml unmarshals this to a Python dict with one key. For now I think it is best to stick with this since introducing Yet Another Change (tm) may introduce more delays which we want to avoid.

Reply:
And thanks! That got me thinking and I discovered a bug in the mock_controller. I omitted the space " " in front of the Timeout keyword which explains the {"OK": None} dict. I'll fix this now in a new story.